libs/cardano-ledger-binary/testlib/Test/Cardano/Ledger/Binary/TreeDiff.hs

```diff
@@ -54,6 +65,14 @@ import Test.Tasty.HUnit (Assertion, assertFailure)
 trimExprViaShow :: Show a => Int -> a -> Expr
 trimExprViaShow _n x = defaultExprViaShow x -- App (take n (drop 1 (show x)) ++ "..") []
 
+tableDoc :: Maybe (Doc AnsiStyle) -> [(String, Doc AnsiStyle)] -> Doc AnsiStyle
+tableDoc mTitle rows =
```

Comment:
`maximum` is a partial function that won't give you much hint of where the exception came from.
```suggestion
tableDoc :: HasCallStack => Maybe (Doc AnsiStyle) -> [(String, Doc AnsiStyle)] -> Doc AnsiStyle
tableDoc mTitle [] = error "`tableDoc` expects a non-empty list"
tableDoc mTitle rows =
```
Alternatively you could require `NonEmpty` instead of a list. More type safety :wink: 
Feel free to decide on which approach to choose with this helper test function

Reply:
I've used `foldr max 0` instead, and `hlint` showed me I could fuse that with the `map (length . fst)` to give `foldr (max . length . fst) 0`